its_hub/algorithms/planning_wrapper.py

```diff
@@ -0,0 +1,303 @@
+        result = PlanningWrappedResult(
+            plan=plan,
+            approaches=approaches,
+            approach_results=approach_results,
+            approach_budgets=approach_budgets,
+            combined_responses=combined_responses,
+            best_approach=best_approach,
+            best_approach_result=best_result,
+        )
+        
+        return result.the_one if return_response_only else result
+    
+    def _select_best_approach(
+        self, approach_results: Dict[str, AbstractScalingResult]
+    ) -> tuple[str, AbstractScalingResult]:
+        """Select the best approach based on algorithm-specific criteria."""
+        
+        # Default: select based on highest confidence/score if available
+        best_approach = None
+        best_result = None
+        best_score = float('-inf')
+        
+        for approach, result in approach_results.items():
+            score = self._get_result_score(result)
+            
+            if score > best_score:
+                best_score = score
+                best_approach = approach
+                best_result = result
+        
+        # Fallback to first approach if no scoring available
+        if best_approach is None:
+            best_approach = list(approach_results.keys())[0]
+            best_result = approach_results[best_approach]
+        
+        return best_approach, best_result
+    
+    def _get_result_score(self, result: AbstractScalingResult) -> float:
+        """Extract a score from the algorithm result for comparison."""
+        
+        # Try different score attributes that algorithms might have
+        score_attrs = [
+            'best_score', 'max_score', 'score', 
+            'confidence', 'probability', 'weight'
+        ]
+        
+        for attr in score_attrs:
+            if hasattr(result, attr):
+                score_val = getattr(result, attr)
+                if isinstance(score_val, (int, float)):
+                    return float(score_val)
+                elif isinstance(score_val, list) and score_val:
+                    return float(max(score_val))
+        
+        # Try to get scores from response collections
+        if hasattr(result, 'scores') and result.scores:
+            return float(max(result.scores))
+        elif hasattr(result, 'all_scores') and result.all_scores:
+            return float(max(result.all_scores))
+        elif hasattr(result, 'log_weights_lst') and result.log_weights_lst:
+            # For ParticleFiltering
+            all_weights = []
+            for weights in result.log_weights_lst:
+                all_weights.extend(weights)
+            return float(max(all_weights)) if all_weights else 0.0
+        
+        # Fallback: use response length as a proxy (longer = more detailed)
+        response = str(result.the_one)
+        return float(len(response)) / 1000.0  # Normalize to reasonable range
+
+
+# Convenience functions for common combinations
```

Comment:
i think these convenience funcs are overkills.
you don't gain any convenience by using them and because it reduces transparency it may make users harder to switch between stuff.